bin/plugin/commands/performance.js

```diff
@@ -214,24 +196,47 @@ async function runPerformanceTests( branches, options ) {
 	// 1- Preparing the tests directory.
 	log( '\n>> Preparing the tests directories' );
 	log( '    >> Cloning the repository' );
-	const baseDirectory = await git.clone( config.gitRepositoryURL );
+
+	/**
+	 * @type {string[]} git refs against which to run tests;
+	 *                  could be commit SHA, branch name, tag, etc...
+	 */
+	if ( branches.length < 2 ) {
+		throw new Error( `Need at least two git refs to run` );
+	}
+
+	const baseDirectory = getRandomTemporaryPath();
+	fs.mkdirSync( baseDirectory, { recursive: true } );
+
+	// @ts-ignore
+	const git = SimpleGit( baseDirectory );
+	await git
+		.raw( 'init' )
+		.raw( 'remote', 'add', 'origin', config.gitRepositoryURL );
+
+	for ( const branch of branches ) {
+		await git.raw( 'fetch', '--depth=1', 'origin', branch );
+	}
+
+	await git.raw( 'checkout', branches[ 0 ] );
```

Comment:
How is the logic here different than the `git clone` call that was removed?

Reply:
one why reason I pulled out the custom `git` abstraction was because I found it hard to follow what was actually going on by jumping around the various functions and modules,  and then I'm not comfortable making changes to the abstraction given my limited understanding of what else relies on it.

the big change here is that all fetches are shallow and that we're not passing `--no-single-branch`. I'm guessing, because there's no record in the codebase for why  `--no-single-branch` was added, that someone added it when one of the referenced branches didn't appear in the shallow clone. however, that option is going to pull in around 1700 commits right now based on the repository today and that amounts to a jump from roughly 20MB downloaded in the clone for what we need to 190MB for all the other things we don't. that also accounts for several minutes of runtime for the job.

later in #45327 (where I'm collecting the set of successful experiments) I'd like to first fetch from the copy of the repo already cloned in previous step and avoid making network calls during the job run. this work alone accounts for around six minutes of runtime and cuts in half the variability of runtime duration in half.

because the `git` abstraction creates directories on clone, because it creates its own `git` object on every call, and because it doesn't return that `git` object we can't easily make minor adjustments in how we are using `git`, which has made these kinds of changes hard to setup or code without pulling in `simple-git` anyway in addition to the abstraction.

Reply:
one why reason I pulled out the custom `git` abstraction was because I found it hard to follow what was actually going on by jumping around the various functions and modules,  and then I'm not comfortable making changes to the abstraction given my limited understanding of what else relies on it.

the big change here is that all fetches are shallow and that we're not passing `--no-single-branch`. I'm guessing, because there's no record in the codebase for why  `--no-single-branch` was added, that someone added it when one of the referenced branches didn't appear in the shallow clone. however, that option is going to pull in around 1700 commits right now based on the repository today and that amounts to a jump from roughly 20MB downloaded in the clone for what we need to 190MB for all the other things we don't. that also accounts for several minutes of runtime for the job.

later in #45327 (where I'm collecting the set of successful experiments) I'd like to first fetch from the copy of the repo already cloned in previous step and avoid making network calls during the job run. this work alone accounts for around six minutes of runtime and cuts in half the variability of runtime duration in half.

because the `git` abstraction creates directories on clone, because it creates its own `git` object on every call, and because it doesn't return that `git` object we can't easily make minor adjustments in how we are using `git`, which has made these kinds of changes hard to setup or code without pulling in `simple-git` anyway in addition to the abstraction.

Reply:
Should we remove it entirely. I guess it's still used by one other command?

Reply:
I would be happy to remove it entirely, as I think that its main contribution is a layer of abstraction that adds indirection while failing to aid in comprehension or efficiency. Generally I'm hesitant to remove things for which I don't fully understand their use or motivation, but I could extend this PR to remove it from the other place also.

Reply:
removed in 0445494162c

it's a bigger change than I prefer, and I plan to add some explanatory comments on things that stood out to me while refactoring (such as why we're using `git pull` to "rest" a branch)

I will request a more careful review than normal to make sure I didn't change anything in the port; already I caught myself accidentally using `trunk` instead of `npmBranchName` in one place.